gateway-service/src/main/java/org/zowe/apiml/gateway/security/mapping/X509NativeMapper.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+
+package org.zowe.apiml.gateway.security.mapping;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.stereotype.Component;
+import org.zowe.apiml.gateway.security.service.schema.source.AuthSource;
+import org.zowe.apiml.gateway.security.service.schema.source.X509AuthSource;
+import org.zowe.commons.usermap.CertificateResponse;
+
+import java.security.cert.CertificateEncodingException;
+import java.security.cert.X509Certificate;
+
+@Slf4j
+@RequiredArgsConstructor
+@Component("x509Mapper")
+@ConditionalOnExpression("'${apiml.security.x509.enabled:false}' == 'true' && '${apiml.security.useInternalMapper:false}' == 'true'")
+public class X509NativeMapper implements AuthenticationMapper {
```

Comment:
there needs to be more debug logs in this class

Reply:
Will add more logs ... Currently, the same amount of log messages is in the X509ExternalMapper. I'll Update the external mapper as well.